src/gt4py/next/ffront/foast_to_itir.py

```diff
@@ -295,23 +295,56 @@ def visit_Compare(self, node: foast.Compare, **kwargs: Any) -> itir.FunCall:
         return self._map(node.op.value, node.left, node.right)
 
     def _visit_shift(self, node: foast.Call, **kwargs: Any) -> itir.Expr:
-        match node.args[0]:
-            case foast.Subscript(value=foast.Name(id=offset_name), index=int(offset_index)):
-                shift_offset = im.shift(offset_name, offset_index)
-            case foast.Name(id=offset_name):
-                return im.lifted_neighbors(str(offset_name), self.visit(node.func, **kwargs))
-            case foast.Call(func=foast.Name(id="as_offset")):
-                func_args = node.args[0]
-                offset_dim = func_args.args[0]
-                assert isinstance(offset_dim, foast.Name)
-                shift_offset = im.shift(
-                    offset_dim.id, im.deref(self.visit(func_args.args[1], **kwargs))
-                )
-            case _:
-                raise FieldOperatorLoweringError("Unexpected shift arguments!")
-        return im.lift(im.lambda_("it")(im.deref(shift_offset("it"))))(
-            self.visit(node.func, **kwargs)
-        )
+        shift_offsets = []
+        for i in range(len(node.args)):
+            match node.args[i]:
+                case foast.Subscript(value=foast.Name(id=offset_name), index=int(offset_index)):
+                    shift_offsets.append(im.shift(offset_name, offset_index))
+                case foast.BinOp(
+                    op=dialect_ast_enums.BinaryOperator.ADD
+                    | dialect_ast_enums.BinaryOperator.SUB,
+                    left=foast.Name(id=dimension),
+                    right=foast.Constant(value=offset_index),
+                ):
+                    if node.args[i].op == dialect_ast_enums.BinaryOperator.SUB:  # type: ignore[attr-defined] # ensured by pattern
+                        offset_index *= -1
+                    shift_offsets.append(im.shift(f"{dimension}off", offset_index))
+                case foast.BinOp(
+                    op=dialect_ast_enums.BinaryOperator.ADD
+                    | dialect_ast_enums.BinaryOperator.SUB,
+                    left=foast.BinOp(
+                        op=dialect_ast_enums.BinaryOperator.ADD
+                        | dialect_ast_enums.BinaryOperator.SUB,
+                        left=foast.Name(id=dimension),
+                        right=foast.Constant(value=offset_index_left),
+                    ),
+                    right=foast.Constant(value=offset_index),
+                ):
+                    if node.args[i].op == dialect_ast_enums.BinaryOperator.SUB:  # type: ignore[attr-defined] # ensured by pattern
+                        offset_index *= -1
+                    shift_offsets.append(
+                        im.shift(f"{dimension}off", offset_index_left + offset_index)
+                    )
```

Comment:
This is for the `IDim+1+1` case right? Let's remove it for now since it is not generic anyway (i.e. doesn't work for `IDim+1+1+1`).

Reply:
As discussed, I removed all sopport for (IDim +1) + 1 again and raise an error in `type_deduction.py` instaed.